Auth0/CredentialsManager.swift

```diff
@@ -186,14 +186,28 @@ public struct CredentialsManager {
                                                  refreshToken: credentials.refreshToken ?? refreshToken,
                                                  expiresIn: credentials.expiresIn,
                                                  scope: credentials.scope)
-                _ = self.store(credentials: newCredentials)
-                callback(nil, newCredentials)
+                if self.willExpire(newCredentials, within: minTTL) {
+                    // TODO: On the next major add a new case to CredentialsManagerError
+                    let error = NSError(domain: "The lifetime of the renewed Access Token is less than minTTL", code: -99999, userInfo: nil)
+                    callback(.failedRefresh(error), nil)
+                } else {
+                    _ = self.store(credentials: newCredentials)
+                    callback(nil, newCredentials)
+                }
             case .failure(let error):
                 callback(.failedRefresh(error), nil)
             }
         }
     }
 
+    func willExpire(_ credentials: Credentials, within ttl: Int) -> Bool {
```

Comment:
Adding a new method for this can be avoided. There's already a `hasExpired` method that behaves exactly like this but without considering the `minTtl`. Let's do that instead.

Reply:
Checking again, this method looks package-private, not public. Then you might want to add the `minTtl` argument to the public one and invoke this -> `hasValid()`

Reply:
It's package-private so it can be unit-tested. It performs a distinct check, so it may be best for it to be in a separate method.